app/src/main/java/org/mozilla/tv/firefox/architecture/KillswitchLinearLayout.kt

```diff
@@ -65,13 +65,16 @@ class KillswitchLinearLayout : LinearLayout {
             return super.setVisibility(View.GONE)
         }
 
-        val currentLocale = LocaleManager.getInstance().getCurrentLocale(context)
         val allAllowed = allowedInLocales == KillswitchLocales.All
         val allowedLocales = (allowedInLocales as? KillswitchLocales.ActiveIn)?.locales
-        val currentLocaleTag = currentLocale?.toLanguageTag()
+        val currentLocale = LocaleManager.getInstance().getCurrentLocale(context)
 
-        val currentLocaleIsAllowed = (allowedLocales != null && currentLocaleTag != null &&
-            allowedLocales.contains(currentLocaleTag)) //TODO fix comparison
+        val currentLocaleIsAllowed = (allowedLocales != null &&
+            allowedLocales.any { allowed ->
+                allowed.language == currentLocale.language &&
```

Comment:
nit: this logic is hard to read. Can we add an extension function to equate them? e.g. `Locale.equalsLanguageAndCountry()`